napari/plugins/manager.py

```diff
@@ -149,6 +149,48 @@ def _register_module(self, plugin_name: str, module_name: str) -> None:
         except Exception as exc:
             raise PluginRegistrationError(plugin_name, module_name) from exc
 
+    def set_implementation_enabled(
+        self, hook_spec: str, plugin_name: str, enabled: bool
+    ) -> None:
```

Comment:
i added those methods (and moved the whole thing to the _HookCaller class), but probably won't be using them internally yet, since this function is mostly there to receive Qt-style signals that (for instance) return the check state of a checkbox...
it's clunkier to do `enable_implementation() if checked else disable_implementation()`
rather than `set_enabled(checked)`